packages/react/src/ui/DatePickerPopup/OneDatePickerPopup.tsx

```diff
@@ -0,0 +1,159 @@
+import { Button } from "@/components/Actions/Button"
+import {
+  GranularityDefinitionKey,
+  OneCalendar,
+  granularityDefinitions,
+} from "@/experimental/OneCalendar"
+import { DateRange } from "@/experimental/OneCalendar/types"
+import { ChevronLeft } from "@/icons/app"
+import { Popover, PopoverContent, PopoverTrigger } from "@/ui/popover"
+import { useMemo, useState } from "react"
+import { GranularitySelector } from "./components/GranularitySelector"
+import { PresetList } from "./components/PresetList"
+import { DatePickerValue, DatePreset } from "./types"
+export interface OneDatePickerPopupProps {
+  onSelect?: (value: DatePickerValue | undefined) => void
+  value?: DatePickerValue
+  defaultValue?: DatePickerValue
+  presets?: DatePreset[]
+  granularities?: GranularityDefinitionKey[]
+  minDate?: Date
+  maxDate?: Date
+  disabled?: boolean
+  hideGoToCurrent?: boolean
+  children: React.ReactNode
+  open?: boolean
+  onOpenChange?: (open: boolean) => void
+}
+
+const presetCustom = "__custom__"
```

Comment:
I don't think we have an agreement, I'd maybe had it uppercase

Reply:
//const + UPPER_SNAKE_CASE = value cannot be reassigned, changed, but immutable

Reply:
done